terraform/applications.tf

```diff
@@ -0,0 +1,111 @@
+resource "juju_application" "kafka" {
+  depends_on = [juju_machine.kafka]
+  model      = juju_model.kafka.name
+
+  units = var.kafka.units
+
+  charm {
+    name    = "kafka"
+    channel = var.kafka.channel
+    series  = var.kafka.series
+  }
+
+  placement = join(",", [for machine in juju_machine.kafka : split(":", machine.id)[1]])
```

Comment:
While it's relatively clear to someone who knows HCL, I'd probably add a comment to lines like this to explain the transformation.